Modules/Filtering/ImageGrid/test/itkBasicArchitectureTest.cxx

```diff
@@ -165,7 +165,11 @@ itkBasicArchitectureTest(int, char *[])
   shrink->Update();
 
   // test RemoveObserver code
-  shrink->RemoveObserver(tag);
+  shrink->RemoveObserver(tag1);
+
+  // test RemoveObserver const code
+  const auto & constShrink = *shrink;
+  constShrink.RemoveObserver(tag2);
```

Comment:
Would like to see an "assert" after the `RemoveObserver` call, to test the effect of the `RemoveObserver` call.

Reply:
no worries, @N-Dekker . I'll add in the assert to get the ball rolling!

Reply:
Thanks Piyush! Your test looks fine to me now 👍 

Honestly I had to double-check the logic of `HasObserver` to see that it's OK! Technically, each `itk::StartEvent()` call creates a _new_ event object, so initially I thought that `HasObserver(itk::StartEvent())` would always return false. But now I see, `HasObserver` checks the runtime-type of its argument, so it makes sense to pass it a newly created object  👍 

By the way, for a new unit test, I usually recommend using GoogleTest, but I see, we don't have any GTest unit test yet for `itk::Object`, as there is no "itkObjectGTest.cxx".  So it's OK to me now to use the old style "itkBasicArchitectureTest.cxx". Especially now that it does Arrange, Act, Assert! 😃

Reply:
> Technically, each itk::StartEvent() call creates a new event object, so initially I thought that HasObserver(itk::StartEvent()) would always return false. But now I see, HasObserver checks the runtime-type of its argument, so it makes sense to pass it a newly created object 👍

Thanks for the deep dive, Niels. I should confess I didn't look into the impl and just went ahead according to the signature - it expected a `const EventObject &` , so sending in a new one _should be fine_ 😬

Reply:
You might have checked that `HasObserver(itk::StartEvent())` still returns true _before_ calling `RemoveObserver(itk::StartEvent())` (while it should return false afterwards), but I think it's good enough already now. I don't want to be over-asking  😺

Reply:
I've added extra checks after the `AddObserver` API calls, thanks for pointing that out! :D